Mathlib/Analysis/Analytic/Meromorphic.lean

```diff
@@ -310,6 +327,82 @@ lemma id {U : Set 𝕜} : MeromorphicOn id U := fun x _ ↦ .id x
 lemma const (e : E) {U : Set 𝕜} : MeromorphicOn (fun _ ↦ e) U :=
   fun x _ ↦ .const e x
 
+/-- The set where a meromorphic function has infinite order is clopen in its domain of meromorphy.
+-/
+theorem isClopen_setOf_order_eq_top {U : Set 𝕜} [CompleteSpace E] (hf : MeromorphicOn f U) :
+    IsClopen { u : U | (hf u.1 u.2).order = ⊤ } := by
+  constructor
+  · rw [← isOpen_compl_iff, isOpen_iff_forall_mem_open]
+    intro z hz
+    rcases (hf z.1 z.2).eventually_eq_zero_or_eventually_ne_zero with h | h
+    · -- Case: f is locally zero in a punctured neighborhood of z
+      rw [← (hf z.1 z.2).order_eq_top_iff] at h
+      tauto
+    · -- Case: f is locally nonzero in a punctured neighborhood of z
+      obtain ⟨t', h₁t', h₂t', h₃t'⟩ := eventually_nhds_iff.1 (eventually_nhdsWithin_iff.1
+        (h.and (hf z.1 z.2).eventually_analyticAt))
+      use Subtype.val ⁻¹' t'
+      constructor
+      · intro w hw
+        simp only [Set.mem_compl_iff, Set.mem_setOf_eq]
+        by_cases h₁w : w = z
+        · rwa [h₁w]
+        · have h₂f := (h₁t' w hw) (Subtype.coe_ne_coe.mpr h₁w)
+          simp [h₂f.2.meromorphicAt_order, h₂f.2.order_eq_zero_iff.2 h₂f.1]
+      · exact ⟨isOpen_induced h₂t', h₃t'⟩
+  · apply isOpen_iff_forall_mem_open.mpr
+    intro z hz
+    conv =>
+      arg 1; intro; left; right; arg 1; intro
+      rw [MeromorphicAt.order_eq_top_iff, eventually_nhdsWithin_iff, eventually_nhds_iff]
+    simp only [Set.mem_setOf_eq] at hz
+    rw [MeromorphicAt.order_eq_top_iff, eventually_nhdsWithin_iff, eventually_nhds_iff] at hz
+    obtain ⟨t', h₁t', h₂t', h₃t'⟩ := hz
+    use Subtype.val ⁻¹' t'
+    simp only [Set.mem_compl_iff, Set.mem_singleton_iff, isOpen_induced h₂t', Set.mem_preimage,
+      h₃t', and_self, and_true]
+    intro w hw
+    simp only [Set.mem_setOf_eq]
+    -- Trivial case: w = z
+    by_cases h₁w : w = z
+    · rw [h₁w]
+      tauto
+    -- Nontrivial case: w ≠ z
+    use t' \ {z.1}, fun y h₁y _ ↦ h₁t' y (Set.mem_of_mem_diff h₁y) (Set.mem_of_mem_inter_right h₁y)
+    constructor
+    · exact h₂t'.sdiff isClosed_singleton
+    · apply (Set.mem_diff w).1
+      exact ⟨hw, Set.mem_singleton_iff.not.1 (Subtype.coe_ne_coe.2 h₁w)⟩
+
+/-- On a connected set, there exists a point where a meromorphic function `f` has finite order iff
+`f` has finite order at every point. -/
+theorem exists_order_ne_top_iff_all_order_ne_top {U : Set 𝕜} [CompleteSpace E]
```

Comment:
```suggestion
theorem exists_order_ne_top_iff_forall {U : Set 𝕜} [CompleteSpace E]
```
I'd drop the end of the lemma name, because what else could it be?